torch/_dynamo/variables/builder.py

```diff
@@ -1263,6 +1269,9 @@ def __eq__(self, other: object) -> bool:
 # Performs automatic dynamic dim determination.
 # Returns tuple of (dynamic_dims, constraint_dims) where each is either a list of dims or None.
 def _automatic_dynamic(e, tx, name, static_shapes):
+    if static_shapes:
+        return [DimDynamic.STATIC] * e.dim(), [None] * e.dim()
```

Comment:
is this policy smearing out further and further?

Reply:
I think there's a refactor I want to do, but I'll probably do it simultaneously with more fine grained stride/storage offset control.